_query-dsl/query-dsl/compound/function-score.md

```diff
@@ -0,0 +1,827 @@
+- Custom:
+    - `script_score`: Uses a script to score documents.
+
+## The weight function
+
+When you use the `weight` function, the original relevance score is multiplied by the floating-point value of `weight`:
+
+```json
+GET shakespeare/_search
+{
+  "query": {
+    "function_score": {
+      "weight": "2"
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+Unlike the `boost` value, the `weight` function is not normalized.
+
+## The random score function
+
+The `random_score` function provides a random score that is consistent for a single user but different between users. The score is a floating-point number in the [0, 1) range. By default, the `random_score` function uses internal Lucene document IDs as seed values, making random values irreproducible because documents can be renumbered after merges. To achieve consistency in generating random values, you can provide `seed` and `field` parameters. The `field` must be a field for which `fielddata` is enabled (commonly, a numeric field). The score is calculated based on the `seed`, the `fielddata` values for the `field`, and a salt calculated based on the index name and shard ID. Because the index name and shard ID are the same for documents that reside in the same shard, documents with the same `field` values will be assigned the same score. To ensure different scores for all documents in the same shard, use a `field` that has unique values for all documents. One option is to use the `_seq_no` field. However, if you choose this field, the scores can change if the document is updated because of the corresponding `_seq_no` update.
+
+The following query uses the `random_score` function with a `seed` and `field`:
+
+```json
+GET blogs/_search
+{
+  "query": {
+    "function_score": {
+      "random_score": {
+        "seed": 20,
+        "field": "_seq_no"
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+## The field value factor function
+
+The `field_value_factor` function recalculates the score using the value of the specified document field. If the field is a multi-valued field, only its first value is used for calculations and the others are not considered. 
+
+The `field_value_factor` function supports the following options:
+
+- `field`: The field to use in score calculations.
+
+- `factor`: And optional factor by which the field value is multiplied. Default is 1.
+
+- `modifier`: One of the modifiers to apply to the field value $$v$$. The following table lists all supported modifiers.
+    
+    Modifier | Formula | Description
+    :--- | :--- | :---
+    `log`| $$\log v$$ | Take the base-10 logarithm of the value. Taking a logarithm of non-positive numbers is an illegal operation and will result in an error. For values between 0 (exclusive) and 1 (inclusive), this function returns non-negative values that will result in an error.  We recommend using `log1p` or `log2p` instead of `log`.
```

Comment:
```suggestion
    `log`| $$\log v$$ | Take the base-10 logarithm of the value. Taking a logarithm of non-positive numbers is an illegal operation and will result in an error. For values between 0 (exclusive) and 1 (inclusive), this function returns non-negative values that will result in an error. We recommend using `log1p` or `log2p` instead of `log`.
```